### ISSUE TITLE
unable to publish even simple LUIS bots in bot composer V1.1.1

### ISSUE THREAD

**Issue by philip-mc** [microsoft/BotFramework-Composer#4496]
Hello there, my partner and I have run into an issue when trying to deploy our first bots.

After following the tutorial on Microsoft's web page (to make a weather bot) we have been unable to get it to publish. The Luis account works in composers emulator, but when we try to push in publishing it comes up with the error.

The log shows the same (or similar) error:

`Publish starting...
{
  "status": "DEPLOY_INFO",
  "message": "C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\interuption\\Cancel.en-us.lu loaded\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\interuption\\GetWeather.en-us.lu loaded\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\interuption\\help.en-us.lu loaded\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\interuption\\weatherbot.en-us.lu loaded\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\interuption\\weatherbot.en-us.lu no changes\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\weatherbot.en-us.lu.dialog\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\weatherbot.lu.dialog\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\Cancel.lu.qna.dialog\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\GetWeather.lu.qna.dialog\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\help.lu.qna.dialog\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\weatherbot.lu.qna.dialog\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "Writing to C:\\Users\\Otter\\AppData\\Roaming\\BotFrameworkComposer\\publishBots\\5fc68681cb5057b653b63f12dd99d431\\azurefunctions\\ComposerDialogs\\generated\\luis.settings..westus.json\n"
}
{
  "status": "DEPLOY_INFO",
  "message": "lubuild succeed"
}
{
  "status": "DEPLOY_INFO",
  "message": "start publish luis"
}
{
  "status": "DEPLOY_ERROR",
  "message": "{\"name\":\"StatusCodeError\",\"statusCode\":400,\"message\":\"400 - \\\"{\\\\r\\\\n  \\\\\\\"error\\\\\\\": {\\\\r\\\\n    \\\\\\\"code\\\\\\\": \\\\\\\"BadArgument\\\\\\\",\\\\r\\\\n    \\\\\\\"message\\\\\\\": \\\\\\\"The source completed without providing data to receive.\\\\\\\"\\\\r\\\\n  }\\\\r\\\\n}\\\"\",\"error\":\"{\\r\\n  \\\"error\\\": {\\r\\n    \\\"code\\\": \\\"BadArgument\\\",\\r\\n    \\\"message\\\": \\\"The source completed without providing data to receive.\\\"\\r\\n  }\\r\\n}\",\"options\":{},\"response\":{\"statusCode\":400},\"stack\":\"StatusCodeError: 400 - \\\"{\\\\r\\\\n  \\\\\\\"error\\\\\\\": {\\\\r\\\\n    \\\\\\\"code\\\\\\\": \\\\\\\"BadArgument\\\\\\\",\\\\r\\\\n    \\\\\\\"message\\\\\\\": \\\\\\\"The source completed without providing data to receive.\\\\\\\"\\\\r\\\\n  }\\\\r\\\\n}\\\"\\n    at new StatusCodeError (C:\\\\Program Files\\\\Bot Framework Composer\\\\resources\\\\app.asar.unpacked\\\\build\\\\plugins\\\\azurePublish\\\\node_modules\\\\request-promise-core\\\\lib\\\\errors.js:32:15)\\n    at Request.plumbing.callback (C:\\\\Program Files\\\\Bot Framework Composer\\\\resources\\\\app.asar.unpacked\\\\build\\\\plugins\\\\azurePublish\\\\node_modules\\\\request-promise-core\\\\lib\\\\plumbing.js:104:33)\\n    at Request.RP$callback [as _callback] (C:\\\\Program Files\\\\Bot Framework Composer\\\\resources\\\\app.asar.unpacked\\\\build\\\\plugins\\\\azurePublish\\\\node_modules\\\\request-promise-core\\\\lib\\\\plumbing.js:46:31)\\n    at Request.self.callback (C:\\\\Program Files\\\\Bot Framework Composer\\\\resources\\\\app.asar.unpacked\\\\build\\\\plugins\\\\azurePublish\\\\node_modules\\\\request\\\\request.js:185:22)\\n    at Request.emit (events.js:210:5)\\n    at Request.<anonymous> (C:\\\\Program Files\\\\Bot Framework Composer\\\\resources\\\\app.asar.unpacked\\\\build\\\\plugins\\\\azurePublish\\\\node_modules\\\\request\\\\request.js:1154:10)\\n    at Request.emit (events.js:210:5)\\n    at IncomingMessage.<anonymous> (C:\\\\Program Files\\\\Bot Framework Composer\\\\resources\\\\app.asar.unpacked\\\\build\\\\plugins\\\\azurePublish\\\\node_modules\\\\request\\\\request.js:1076:12)\\n    at Object.onceWrapper (events.js:299:28)\\n    at IncomingMessage.emit (events.js:215:7)\\n    at endReadableNT (_stream_readable.js:1183:12)\\n    at processTicksAndRejections (internal/process/task_queues.js:80:21)\"}"
}
400 - "{\r\n  \"error\": {\r\n    \"code\": \"BadArgument\",\r\n    \"message\": \"The source completed without providing data to receive.\"\r\n  `}\r\n}"`

We figured this may be some sort of access issue to azure, but have tried from different access tokens (both generated in azure CLI and Azure Cloud Shell) and from both users (both have owner privileges) but this is an error we receive if we try to provision new resources, or use existing ones. 

We have managed to publish bots using regex with no issues, it seem to be any we do with LUIS any help to shed light on this matter would be greatly appreciated!

**Comment by stevengum** [comment]
@dmvtech can you investigate this issue? Thanks!

**Comment by dmvtech** [comment]
@philip-mc Do you have both authoring and prediction resources setup in Azure already? And are you providing keys for both of those in the publishing settings? Also be sure to make sure the LUIS region is correct.

**Comment by philip-mc** [comment]
hello and thanks for getting in touch. We have both authoring and prediction keys set up in azure both through manual npm install, node composer provisioner as per the tutorials as well as doing so manually as well. We have also tried a few different regions, Us west was the original default, but based on some contact with others in similar situations we did also try setting everything up in the same region as our tenant and still not working. we have tried generating access tokens through both my partner (azure owner) and myself (admin with privileges) and neither work. 
We can do regex but as soon as we try for Luis it stops working. 

Emulation works fine on local, contacting Luis and replying via Luis trace v3

**Comment by dmvtech** [comment]
@philip-mc Thank you. Very strange. Not sure what is going and I'm not able to reproduce.

@luhan2017 @cwhitten @tdurnford Any ideas here? I'm not familiar with the error `the source completed without providing data to receive` or what could causing it. Anything else to check for?

**Comment by luhan2017** [comment]
@philip-mc I've replied you through #4395, this error is caused by this API, https://westus.dev.cognitive.microsoft.com/docs/services/5890b47c39e2bb17b84a55ff/operations/5be313cec181ae720aa2b26c/console, if the access-token and authoring key are both correct, and it still not works, could you please help submit a ICM ticket to luis team referring to this doc? https://docs.microsoft.com/en-us/azure/azure-portal/supportability/how-to-create-azure-support-request. It is the best way to get their attention and have them prioritized for this.

**Comment by dmvtech** [comment]
I am closing this issue as it is related to LUIS and not Composer. Feel free to reopen if you determine that is not the case.

**Comment by jpd21122012** [comment]
@philip-mc you were able to contact support?

**Comment by philip-mc** [comment]
yes, sorry for the long delay we have run into further errors non-related to this direct issue. Contacting the luis team (we had to upgrade to development help and support to send a ticket), they proceeded to trace our attempts to gather a token and it turns out all tokens where trying to generate through an old legacy subscription despite setting the subscription!

Upon further investigation it seems the old subscription we had "Access to azure active directory" was broken and we had no permissions to do anything with it. finding this link here - https://www.jasonfritts.me/2020/04/07/what-is-the-access-to-azure-active-directory-subscription-for/ we found it could possibly be removed from the subscription by contacting Microsoft support. 

Whilst we await the subscriptions removal, I found that dropping my azure admin privileges to a contributor took me off the legacy tenant. For people with only one admin, possibly creating an extra account with 'contributor' and only access to the subscription you want may be of help, as it certainly was for us.

By having a user without access to the broken subscription we where able to generate an arm token and successfully publish a bot on azure.
